include/eve/detail/function/simd/x86/friends.hpp

```diff
@@ -146,11 +146,23 @@ namespace eve::detail
         else  if constexpr(use_avx2 && c == category::uint16x16)  return _mm256_cmpeq_epi16(v,w);
         else  if constexpr(use_avx2 && c == category::int8x32  )  return _mm256_cmpeq_epi8 (v,w);
         else  if constexpr(use_avx2 && c == category::uint8x32 )  return _mm256_cmpeq_epi8 (v,w);
-        else  if constexpr( c == category::int64x2  )             return map(eq,v,w);
+        else  if constexpr( c == category::int64x2  )
+        {
+          if constexpr(N::value == 2)
+            return as_logical_t<wide<T, N>>{ v.get(0) == w.get(0), v.get(1) == w.get(1) };
+          else
+            return as_logical_t<wide<T, N>>{ v.get(0) == w.get(0) };
+        }
```

Comment:
Why is it better than just calling map?

Reply:
g++ has issue with this one map calls, this version produce the correct code